average.js

```diff
@@ -1,6 +1,7 @@
 
 function average(numbers) {
-  return numbers.reduce((p, c)=> p + c, 0) / numbers.length;
+  const newArr = numbers.filter( value => !Number.isNaN(value) );
+  return newArr.reduce((p, c)=> p + c, 0) / newArr.length;
 }
 
 module.exports = {average};
```

Comment:
Try a new test: The tests in test/average.test.js express the functionality of the average function.
Consider this situation: IoT sensors sometimes report a lot of variation in their readings. If you consider that the sensor is reading out battery temperature, there cannot be sudden variation in reality. Hence, a lot of variation in the readings would mean a defective sensor.
Yet, our `average` function would compute the average as though everything was ok.
Add functionality: Come up with a way to return `NaN` from the `average` function, if the variation is too much.
You will need to:
- determine the meaning of 'too much'. E.g., the difference between max and min, or the difference in successive readings. Pick the method you feel is best. Include any references you used for it.
- implement a test to check that the `average` function returns `NaN` on high variance. Name the test appropriately.
- implement the functionality to pass the test.

Reply:
Great suggestion actually.